Change dates to pointers

puerco opened the pull request:
This PR changes the dates in our vex structs to pointers to be able to nil them

Signed-off-by: Adolfo Garcia Veytia (puerco) <puerco@chainguard.dev>

luhring commented on pkg/vex/statement.go:
```diff
@@ -102,14 +102,14 @@ func SortStatements(stmts []Statement, documentTimestamp time.Time) {
 		iTime := stmts[i].Timestamp
 		if iTime.IsZero() {
-			iTime = documentTimestamp
+			iTime = &documentTimestamp
```
Should this check be `if iTime == nil { ...` now?

(and same thing for `jTime`)

luhring commented on pkg/vex/statement.go:
```diff
@@ -102,14 +102,14 @@ func SortStatements(stmts []Statement, documentTimestamp time.Time) {
 		}
 
-		return iTime.Before(jTime)
+		return iTime.Before(*jTime)
```
We have to be careful here about the possibility of a nil pointer deference

luhring commented on pkg/vex/vex.go:
```diff
@@ -123,7 +124,7 @@ func (vexDoc *VEX) StatementFromID(id string) *Statement {
 func Sort(docs []*VEX) []*VEX {
 	sort.Slice(docs, func(i, j int) bool {
-		return docs[i].Timestamp.Before(docs[j].Timestamp)
+		return docs[i].Timestamp.Before(*(docs[j].Timestamp))
```
We have to be careful about a nil pointer dereference. Maybe there should be a helper method on `vex.VEX` to return its timestamp, which will either be `doc.Timestamp` or it should be the latest value among the statements? Otherwise it's an error condition

puerco commented:
I've added checks to watch out for nil pointers before referencing them. I've also dropped my statement sorting function and merge now uses yours @luhring , which is more complete.

luhring reviewed (approved):
LGTM!
